File: cpp/unittests/ReverseNodesInKGroupTest.cpp
#include "ReverseNodesInKGroup.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(ReverseNodesInKGroup, Solution25_1) {
  auto Head1 = newLinkedList({1, 2, 3, 4, 5});
  const int K1 = 2;
  const auto Expected1 = newLinkedList({2, 1, 4, 3, 5});
  Head1 = Solution25_1().reverseKGroup(Head1, K1);
  EXPECT_EQ(*Expected1, *Head1);
  deleteLinkedList(Head1);
  deleteLinkedList(Expected1);

  auto Head2 = newLinkedList({1, 2, 3, 4, 5});
  const int K2 = 3;
  const auto Expected2 = newLinkedList({3, 2, 1, 4, 5});
  Head2 = Solution25_1().reverseKGroup(Head2, K2);
  EXPECT_EQ(*Expected2, *Head2);
  deleteLinkedList(Head2);
  deleteLinkedList(Expected2);
}

File: java/src/test/java/io/binac/leetcode/ReverseNodesInKGroupTest.java
package io.binac.leetcode;

import io.binac.leetcode.util.LinkedLists;
import io.binac.leetcode.util.ListNode;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ReverseNodesInKGroupTest {
    private static final ReverseNodesInKGroup.Solution1 SOLUTION1 = new ReverseNodesInKGroup.Solution1();

    private final ListNode HEAD1 = LinkedLists.asLinkedList(1, 2, 3, 4, 5);
    private final int K1 = 2;
    private final ListNode EXPECTED1 = LinkedLists.asLinkedList(2, 1, 4, 3, 5);

    private final ListNode HEAD2 = LinkedLists.asLinkedList(1, 2, 3, 4, 5);
    private final int K2 = 3;
    private final ListNode EXPECTED2 = LinkedLists.asLinkedList(3, 2, 1, 4, 5);

    private final ListNode HEAD3 = LinkedLists.asLinkedList(1, 2, 3, 4, 5);
    private final int K3 = 1;
    private final ListNode EXPECTED3 = LinkedLists.asLinkedList(1, 2, 3, 4, 5);

    private final ListNode HEAD4 = LinkedLists.asLinkedList(1);
    private final int K4 = 1;
    private final ListNode EXPECTED4 = LinkedLists.asLinkedList(1);

    @Test
    void test1() {
        assertTrue(LinkedLists.equals(EXPECTED1, SOLUTION1.reverseKGroup(HEAD1, K1)));
        assertTrue(LinkedLists.equals(EXPECTED2, SOLUTION1.reverseKGroup(HEAD2, K2)));
        assertTrue(LinkedLists.equals(EXPECTED3, SOLUTION1.reverseKGroup(HEAD3, K3)));
        assertTrue(LinkedLists.equals(EXPECTED4, SOLUTION1.reverseKGroup(HEAD4, K4)));
    }
}